### PR TITLE
Fix codebridge sources race condition

### PR DESCRIPTION
**Pull request by molly-moen**
@moshebaricdo found a bug in Web Lab 2 (that also existed in Python Lab), where if you were on a Web Lab 2 level, then go to a Web Lab 2 bubble choice level, one of your files can get overwritten by a file from the old level.

This issue was identical to the one fixed in #69541, which we never successfully root caused. I got closer to a root cause here, although I'm still not 100% sure on the exact reason why.

TL;DR: there was a race condition. It turned out when going from a level (call this level 1) to a bubble choice level (call this level 2), then to a sublevel (call this level 3), the old sources from Level 1 persist briefly after the level properties have changed. We set up the new level with the old sources, including rendering an editor with the old sources. Then the new sources come in and we render a new editor with the new sources. An editor update event then runs on the codemirror side with the old sources. If the id of that file is different from the id of our current active file, and we have a file with that id in the new sources, it gets overwritten. This is because if it were the same as our current file, we would successfully overwrite it with the new sources, but if it's not open, we aren't trying to save the new code.

<details>
<summary>Unsuccessful Attempt</summary>
My first attempt for this was to try to clean up codemirror using the [destroy](https://codemirror.net/docs/ref/#view.EditorView.destroy) method. This didn't stop the update event from running. I'm still not sure why. I didn't keep the change because I wasn't sure of any unintended consequences of adding that cleanup when we weren't before. 
</details>

### Solution(s)
My first fix for this was to only send code change events if the document changed. This fixed the issue, but seemed fragile. I kept this in there because it will reduce load on our redux reducer, which checks for changes before actually changing anyway. This is a faster way to do that check, since the change event has a flag on if there was a change.

The better fix for this was to only render codebridge if we had sources for the current level id in our level properties. I fixed this by tracking the sources level id in the lab2 project redux. 

There may be other solutions here (potentially long-term having unified lab2 sources storage would be one), but this felt lightweight and should fix the problem and any other edge cases where it occurs.

## Testing story
Tested locally.

## PR Creation Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy impacts have been documented
- [ ] Security impacts have been documented
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/weblab2/Weblab2View.tsx [hunk 1]
```diff
@@ -139,7 +142,7 @@ const Weblab2View: React.FC<
 
   return (
     <div className={moduleStyles.weblab2Container}>
-      {hasSource && (
+      {hasSource && sourceLevel === levelProperties.id && (
```
Comment (fisher-alice) [inline apps/src/weblab2/Weblab2View.tsx:145]: To confirm my understanding...
I repro'd error when going from a sublevel with inactive file: 'demo1', then going to a level with active file 'demo1', then going back to parent level (bubble choice) then sublevel. When I open 'demo1' in sublevel editor (make active), it has been overwritten.
However, if I go from sublevel to level which has a file named 'demo1' but it is inactive, and then go back to parent level (bubble choice) then sublevel, the 'demo1' file is NOT overwritten.
This makes sense according to your explanation - only the active file fires an editor update and saves to redux. Thanks so much for the detailed description!

I do have a question ... why does this bug only manifest when going from level to bubble choice (parent) to sublevel and not level to level?

**Review by fisher-alice — APPROVED** [review]
Great sleuthing!

File: apps/src/weblab2/Weblab2View.tsx (hunk 1 above)
Comment (molly-moen) [reply apps/src/weblab2/Weblab2View.tsx:145]: I'm not sure why it only manifests in specific scenarios 🤷‍♀️ My best guess is it's a race condition that happens to trigger in those specific setups.
